plugins/modules/s3_object.py

```diff
@@ -591,49 +604,73 @@ def delete_key(module, s3, bucket, obj):
         s3.delete_object(Bucket=bucket, Key=obj)
         module.exit_json(msg="Object deleted from bucket %s." % (bucket), changed=True)
     except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed while trying to delete %s." % obj)
-
+        raise S3ObjectFailure("Failed while trying to delete %s." % obj, e)
 
-def create_dirkey(module, s3, bucket, obj, encrypt, expiry):
-    if module.check_mode:
-        module.exit_json(msg="PUT operation skipped - running in check mode", changed=True)
+def put_object_acl(module, s3, bucket, obj, params={}):
```

Comment:
You should avoid to default params on a dict like that.

See: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html